backend/src/routers/books.py

```diff
@@ -1,11 +1,43 @@
-from fastapi import APIRouter
+from fastapi import APIRouter, HTTPException
 from src.db import db
-
+from prisma.models import Book, Page
+from typing import Optional, List
+import logging
 
 books_router = APIRouter()
 
 
-@books_router.get("/books/", tags=["books"])
+@books_router.get("/books/search", response_model=List[Book], tags=["books"])
 async def search_books(limit: int = 10):
     books = await db.book.find_many(take=limit, include={"courses": True})
     return books
+
+@books_router.get("/books/{book_id}", response_model=Book, tags=["books"])
+async def get_book(book_id: int):
+    book = await db.book.find_unique(where={"id": book_id}, include={"pages": True})
+    if not book:
+        raise HTTPException(status_code=404, detail="Book not found")
+    return book
+
+logger = logging.getLogger(__name__)
```

Comment:
wanna bump this out to a new file either to main or to a logger.py file kinda like the db?